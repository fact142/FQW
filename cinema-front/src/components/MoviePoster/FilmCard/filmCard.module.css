.card{
    border: solid #ffb8c6 2px;
    border-radius: 10px;
    box-shadow:5px 5px 5px #ffb8c6;
    max-width: 224px;
    min-width: 224px;
    height: 300px;
    display: flex;
    margin: 40px;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    background-color: white;
    padding: 10px;
}
.title{
    font-weight: 600;
    font-family: 'SFMono-Regular', Consolas, 'Liberation Mono', Menlo, Courier, monospace;
}
.info{
    color: rgba(0,0,0,0.5);
    font-size: small;
}
.description{
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 10px;
}
.button{
    background-color: #ffb8c6;
    justify-self: flex-end;
    place-self: center;
    margin: 20px;
}
