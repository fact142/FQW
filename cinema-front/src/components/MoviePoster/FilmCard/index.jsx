import styles from './filmCard.module.css';
import { Button } from 'antd';
import ScheduleModal from '../ScheduleModal';
import { useState } from 'react';

const FilmCard = ({film}) => {
  const [isModalVisible, setIsModalVisible] = useState(false);

  const showModal = () => {
    setIsModalVisible(true);
  };

  const handleOk = () => {
    setIsModalVisible(false);
  };

  const handleCancel = () => {
    setIsModalVisible(false);
  };

  return(
    <>
      <div className={styles.card}>
        <div className={styles.description}>
          <div className={styles.title}>
            {film.title}
          </div>
          <div className={styles.info}>
            {film.country}, {film.year}
          </div>
          <div className={styles.info}>
            {film.duration} мин
          </div>
        </div>
        <Button className={styles.button} onClick={showModal}>
          Расписание
        </Button>
      </div>
      <ScheduleModal
        onOk={handleOk}
        onCancel={handleCancel}
        filmId={film.id}
        visible={isModalVisible}
        filmTitle={film.title}
      />
    </>
  )
}

export default FilmCard;
