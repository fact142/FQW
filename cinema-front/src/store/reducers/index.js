import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import userReducer from './userReducer';
import sessionReducer from './sessionReducer';

const rootReducer = combineReducers({
  user: userReducer,
  session: sessionReducer
});
const composedEnhancer = composeWithDevTools(applyMiddleware(thunk));
const store = createStore(rootReducer, composedEnhancer);

export default store;
