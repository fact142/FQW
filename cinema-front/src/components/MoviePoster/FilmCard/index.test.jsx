import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../store/reducers';
import FilmCard from './index';

const TestingCreatePassword = () => (
  <BrowserRouter>
    <Provider store={store}>
      <FilmCard film={{id: 1, title: 11, duration: 111, year: 2022}}/>
    </Provider>
  </BrowserRouter>
);

describe('CreatePassword form', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <TestingCreatePassword />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render password input', () => {
    const { getByPlaceholderText } = render(
      <TestingCreatePassword />,
    );
    expect(getByPlaceholderText('New password')).toBeInTheDocument();
    expect(getByPlaceholderText('New password')).toBeRequired();
  });
  it('should render confirm password input', () => {
    const { getByPlaceholderText } = render(
      <TestingCreatePassword />,
    );
    expect(getByPlaceholderText('Confirm password')).toBeInTheDocument();
    expect(getByPlaceholderText('Confirm password')).toBeRequired();
  });
  it('should render send button', () => {
    const { getByText } = render(
      <TestingCreatePassword />,
    );
    expect(getByText('Send a link')).toBeInTheDocument();
  });
});

it('should change value when typed', () => {
  const { getByPlaceholderText } = render(
    <TestingCreatePassword />,
  );
  userEvent.type(getByPlaceholderText('New password'), '123456');
  expect(getByPlaceholderText('New password')).toHaveValue('123456');
});
