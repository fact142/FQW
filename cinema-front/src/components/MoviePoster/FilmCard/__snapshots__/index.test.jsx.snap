// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePassword form should renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="description"
  >
    <div
      className="title"
    >
      11
    </div>
    <div
      className="info"
    >
      , 
      2022
    </div>
    <div
      className="info"
    >
      111
       мин
    </div>
  </div>
  <button
    className="ant-btn ant-btn-default button"
    onClick={[Function]}
    type="button"
  >
    <span>
      Расписание
    </span>
  </button>
</div>
`;
